google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PublisherImpl.java

```diff
@@ -90,55 +83,41 @@
   private final Duration sendBundleDeadline;
   private ScheduledFuture<?> currentAlarmFuture;
 
-  PublisherImpl(Builder builder) throws IOException {
-    topic = builder.topic;
-
-    maxBundleMessages = builder.bundlingSettings.getElementCountThreshold();
-    maxBundleBytes = builder.bundlingSettings.getRequestByteThreshold();
-    maxBundleDuration = builder.bundlingSettings.getDelayThreshold();
+  PublisherImpl(String topic, PubSubOptions psOptions, Settings settings) throws IOException {
```

Comment:
Option splitting logic:
- `topic` => we definitely need to ask for it, otherwise we don't know where to publish to
- `psOptions` => we want to infer these most of the time, so that the user can't push/pull to different endpoints by mistake
- `settings` => in simple cases, just use the default, might need to tweak for performance reasons